### PR TITLE
megasync: 4.3.5.0 -> 4.4.0.0

### LINKED ISSUE #112572: MEGAsync - Segmentation fault
**Describe the bug**
Latest MEGAsync app fails to start (Segmentation fault), probably because of `crypto++`.
Not sure if it is NixOS related or just a MEGAsync bug.

**To Reproduce**
1. Install `megasync`
2. Run it
3. Segmentation fault

**Expected behavior**
No crash

**Additional context**
Crash dump:
```
MEGAprivate ERROR DUMP
Application: MEGAsync [64 bit]
Version code: 4305.0
Module name: megasync
Timestamp: 1612899382036
Operating system: Linux
System version:  nixos 21.03pre26/#1-NixOS SMP Wed Feb 3 22:26:02 UTC 2021
System release:  5.4.95
System arch: x86_64
Error info:
Segmentation fault (11) at address 0x20
Stacktrace:
/nix/store/nf9vk0s2w5w9xqw70jdzizbnq8bipr7c-crypto++-8.4.0/lib/libcryptopp.so.8(_ZN8CryptoPP16IteratedHashBaseIjNS_18HashTransformationEE14TruncatedFinalEPhm+0x1c0) [0x7f6011353990]
/nix/store/nf9vk0s2w5w9xqw70jdzizbnq8bipr7c-crypto++-8.4.0/lib/libcryptopp.so.8(_ZN8CryptoPP16IteratedHashBaseIjNS_18HashTransformationEE14TruncatedFinalEPhm+0x1c0) [0x7f6011353990]
/nix/store/nf9vk0s2w5w9xqw70jdzizbnq8bipr7c-crypto++-8.4.0/lib/libcryptopp.so.8(_ZN8CryptoPP10RandomPool18IncorporateEntropyEPKhm+0xda) [0x7f60113a0d7a]
/nix/store/nf9vk0s2w5w9xqw70jdzizbnq8bipr7c-crypto++-8.4.0/lib/libcryptopp.so.8(_ZN8CryptoPP20AutoSeededRandomPool6ReseedEbj+0x43) [0x7f6011387323]
/nix/store/hklsbsd9xyh42x143r7j0ldb2d3bhrbb-megasync-4.3.5.0/bin/megasync() [0x5d7801]
/nix/store/hklsbsd9xyh42x143r7j0ldb2d3bhrbb-megasync-4.3.5.0/bin/megasync() [0x6cfa37]
/nix/store/hklsbsd9xyh42x143r7j0ldb2d3bhrbb-megasync-4.3.5.0/bin/megasync() [0x6d03bb]
/nix/store/hklsbsd9xyh42x143r7j0ldb2d3bhrbb-megasync-4.3.5.0/bin/megasync() [0x669d4f]
/nix/store/hklsbsd9xyh42x143r7j0ldb2d3bhrbb-megasync-4.3.5.0/bin/megasync() [0x7b8265]
/nix/store/hklsbsd9xyh42x143r7j0ldb2d3bhrbb-megasync-4.3.5.0/bin/megasync() [0x4948df]
/nix/store/gafigwfaimlziam6qhw1m8dz4h952g1n-glibc-2.32-35/lib/libc.so.6(__libc_start_main+0xed) [0x7f600f0e7ded]
/nix/store/hklsbsd9xyh42x143r7j0ldb2d3bhrbb-megasync-4.3.5.0/bin/megasync() [0x49663a]
```

**Notify maintainers**

@michojel 

**Metadata**
```
 - system: `"x86_64-linux"`
 - host os: `Linux 5.4.95, NixOS, 21.03pre268835.8c8731330b5 (Okapi)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.10`
 - channels(mateusz): `"home-manager, nixpkgs-21.03pre268835.8c8731330b5"`
 - channels(root): `"home-manager, nixos-21.03pre268835.8c8731330b5, nixos-hardware, nixpkgs-21.03pre268835.8c8731330b5"`
 - nixpkgs: `/home/mateusz/.nix-defexpr/channels/nixpkgs`
```

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute:
# a list of nixos modules affected by the problem
module:
```

### PR DESCRIPTION
**Pull request by michojel**
###### Motivation for this change

Resolves #112572

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 114776` at d41b0ba8 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li><a href="https://gist.github.com/23f640135594ce8d1b13178c7265ef04">megasync</a></li></li>
  </ul>
</details>
<details>
  <summary>1 suggestion:</summary>
  <ul>
    <li>warning: unused-argument

Unused argument: `fetchpatch`.
Near pkgs/applications/misc/megasync/default.nix:10:3:
```
   |
10 | , fetchpatch
   |   ^
```
</li>
  </ul>
</details>

**Comment by michojel** [comment]
> Result of `nixpkgs-review pr 114776` at [d41b0ba]
> ...
> 1 suggestion:

resolved

File: pkgs/applications/misc/megasync/default.nix [hunk 1]
```diff
@@ -25,16 +25,18 @@
 , unzip
 , wget
 }:
-
+let
+  srcFlavor = "Linux";
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/megasync/default.nix:29]: Why are we introducing this? It can't be overwritten right now.

Comment (michojel) [reply pkgs/applications/misc/megasync/default.nix]: I was trying out "Win" flavor and I found it convenient. Nevertheless, using other flavors probably does not make sense as long as "Linux" is available for particular version.
I've just removed this change.

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 114776` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
1 package built:
</summary>
<ul>
<li>megasync</li>
</ul>
</details>

The following issues got detected with the above build packages.
Please fix at least the ones listed with your changed packages:

<details>
megasync:
<details>


warning: build-tools-in-build-inputs
libtool is a build tool so it likely goes to `nativeBuildInputs`, not `buildInputs`.

Near pkgs/applications/misc/megasync/default.nix:42:3:

```
   |
42 |   buildInputs = [
   |   ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/build-tools-in-build-inputs.md
warning: build-tools-in-build-inputs
unzip is a build tool so it likely goes to `nativeBuildInputs`, not `buildInputs`.

Near pkgs/applications/misc/megasync/default.nix:42:3:

```
   |
42 |   buildInputs = [
   |   ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/build-tools-in-build-inputs.md
warning: stale-substitute
Stale substituteInPlace detected.
substituteStream(): WARNING: pattern '/bin/bash' doesn't match anything in file 'src/MEGASync/mega/clean.sh'
See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/stale-substitute.md
warning: stale-substitute
Stale substituteInPlace detected.
substituteStream(): WARNING: pattern '/bin/bash' doesn't match anything in file 'src/MEGASync/mega/bindings/ios/3rdparty/build-cryptopp.sh'
See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/stale-substitute.md
warning: stale-substitute
Stale substituteInPlace detected.
substituteStream(): WARNING: pattern '/bin/bash' doesn't match anything in file 'src/MEGASync/mega/bindings/ios/3rdparty/build-libsodium.sh'
See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/stale-substitute.md
warning: stale-substitute
Stale substituteInPlace detected.
substituteStream(): WARNING: pattern '/bin/bash' doesn't match anything in file 'src/MEGASync/mega/bindings/ios/3rdparty/build-all.sh'
See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/stale-substitute.md
warning: stale-substitute
Stale substituteInPlace detected.
substituteStream(): WARNING: pattern '/bin/bash' doesn't match anything in file 'src/MEGASync/mega/bindings/ios/3rdparty/build-libuv.sh'
See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/stale-substitute.md
warning: stale-substitute
Stale substituteInPlace detected.
substituteStream(): WARNING: pattern '/bin/bash' doesn't match anything in file 'src/MEGASync/mega/bindings/ios/3rdparty/build-cares.sh'
See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/stale-substitute.md
warning: stale-substitute
Stale substituteInPlace detected.
substituteStream(): WARNING: pattern '/bin/bash' doesn't match anything in file 'src/MEGASync/mega/bindings/ios/3rdparty/build-openssl.sh'
See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/stale-substitute.md
warning: stale-substitute
Stale substituteInPlace detected.
substituteStream(): WARNING: pattern '/bin/bash' doesn't match anything in file 'src/MEGASync/mega/bindings/ios/3rdparty/build-webrtc.sh'
See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/stale-substitute.md
warning: stale-substitute
Stale substituteInPlace detected.
substituteStream(): WARNING: pattern '/bin/bash' doesn't match anything in file 'src/MEGASync/mega/bindings/ios/3rdparty/build-mediainfolib.sh'
See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/stale-substitute.md
warning: stale-substitute
Stale substituteInPlace detected.
substituteStream(): WARNING: pattern '/bin/bash' doesn't match anything in file 'src/MEGASync/mega/bindings/ios/3rdparty/build-curl.sh'
See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/stale-substitute.md
warning: stale-substitute
Stale substituteInPlace detected.
substituteStream(): WARNING: pattern '/bin/bash' doesn't match anything in file 'src/MEGASync/mega/bindings/ios/3rdparty/build-libwebsockets.sh'
See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/stale-substitute.md
warning: stale-substitute
Stale substituteInPlace detected.
substituteStream(): WARNING: pattern '/bin/bash' doesn't match anything in file 'src/MEGASync/mega/autogen.sh'
See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/stale-substitute.md
warning: stale-substitute
Stale substituteInPlace detected.
substituteStream(): WARNING: pattern '/bin/bash' doesn't match anything in file 'src/MEGASync/mega/contrib/build_ffmpeg/ffmpeg/build_minimum.sh'
See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/stale-substitute.md
warning: stale-substitute
Stale substituteInPlace detected.
substituteStream(): WARNING: pattern '/bin/bash' doesn't match anything in file 'src/MEGASync/mega/contrib/build_pdfium/build.sh'
See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/stale-substitute.md
warning: stale-substitute
Stale substituteInPlace detected.
substituteStream(): WARNING: pattern '/bin/bash' doesn't match anything in file 'src/MEGASync/mega/contrib/clang-analyzer.sh'
See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/stale-substitute.md
</details>
</details>

**Comment by michojel** [comment]
@SuperSandro2000 thanks for the suggestions.

I addressed the nativeBuildInputs warnings. Regarding the stale-substitutes, I'd like to avoid checking of presence of `/bin/bash` before the substitution. In such case it's a noop AFAIK. What do you think?

I wanted to run the review with the  nixpkgs-review-checks extension myself, but ended up with:
```
# source ~/wsp/up/nixp/nixpkgs-review-checks/bashrc
# nixpkgs-review pr 114776
cached-nix-shell: updating cache
unpacking 'https://github.com/NixOS/nixpkgs/archive/nixpkgs-unstable.tar.gz'...
these paths will be fetched (14.29 MiB download, 61.12 MiB unpacked):
  /nix/store/0a66yh5sflnlnpvd528yld8j1y1am1si-libkrb5-1.18-dev
  /nix/store/0c9qxc8vyyglmjcxbv2gfx55pq7wc53y-bloaty-1.1
 ...
copying path '/nix/store/kjfl3bmyh4mgnikbzw8ryifx8zr0d97g-savepagenow-1.1.1' from 'https://cache.nixos.org'...
cached-nix-shell: "/tmp/nix-9361-1": expected "3896f10787f74542dec12486274a2a74", got "-"
cached-nix-shell: some files are already updated, cache won't be reused
cached-nix-shell: done in 22.611234472s
usage: /nix/store/f3x8n5jvj7qdby52psd6z4nrkgsdb4rw-nixpkgs-review-2.5.0/bin/nixpkgs-review [-h] {post-result,approve,comments,merge,pr,rev,wip} ...
/nix/store/f3x8n5jvj7qdby52psd6z4nrkgsdb4rw-nixpkgs-review-2.5.0/bin/nixpkgs-review: error: unrecognized arguments: --skip-package --skip-package digikam --skip-package envoy --skip-package iosevka --skip-package librealsense --skip-package libreoffice --skip-package lumo --skip-package pcl --skip-package mrtrix --skip-package simpleitk --skip-package smesh --skip-package tts --skip-package-regex .*ceph.* --skip-package-regex freecad --skip-package-regex opencascade.* --skip-package-regex .*pytorch.* --skip-package-regex .*sage.* --skip-package-regex samba4?Full --skip-package-regex .*tensorflow.* --skip-package-regex qgis.* --skip-package-regex .*vtk.* --skip-package-regex .*wine.* --skip-package-regex python\d+Packages\.baselines --skip-package-regex python\d+Packages\.edward --skip-package-regex python\d+Packages\.mask-rcnn --skip-package-regex python\d+Packages\.pyro-ppl --skip-package-regex python\d+Packages\.scikit-tda --skip-package-regex python\d+Packages\.torchgpipe --skip-package-regex python\d+Packages\.torchvision --skip-package-regex python\d+Packages\.tflearn --skip-package-regex python\d+Packages\.umap-learn 114776
```

**Comment by archseer** [comment]
Quick bump, looks like the lint was resolved and this should be LGTM? I'm hoping to see this merged since 4.3.5.0 on master segfaults.

**Comment by michojel** [comment]
squashed and rebased

File: pkgs/applications/misc/megasync/default.nix [hunk 2]
```diff
@@ -24,21 +25,30 @@
...
+      autoconf
+      automake
+      doxygen
+      libtool
+      lsb-release
+      pkg-config
+      qttools
+      swig
+      unzip
+    ];
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/megasync/default.nix:51]:
```suggestion
    autoconf
    automake
    doxygen
    libtool
    lsb-release
    pkg-config
    qttools
    swig
    unzip
  ];
```

File: pkgs/applications/misc/megasync/default.nix [hunk 3]
```diff
@@ -24,21 +25,30 @@
...
   src = fetchFromGitHub {
     owner = "meganz";
     repo = "MEGAsync";
     rev = "v${version}_Linux";
-    sha256 = "0rr1jjy0n5bj1lh6xi3nbbcikvq69j3r9qnajp4mhywr5izpccvs";
+    sha256 = "1xggca7283943070mmpsfhh7c9avy809h0kgmf7497f4ca5zkg2y";
     fetchSubmodules = true;
   };
 
   nativeBuildInputs =
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/megasync/default.nix:40]:
```suggestion
  nativeBuildInputs = [
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]
Please follow our formatting guide.

File: pkgs/applications/misc/megasync/default.nix (hunk 2 above)
Comment (michojel) [reply pkgs/applications/misc/megasync/default.nix]: my bad for relying solely on nix-community/nixpkgs-fmt
addressed

Comment (SuperSandro2000) [reply pkgs/applications/misc/megasync/default.nix]: That shouldn't introduce this change by itself.

Comment (michojel) [reply pkgs/applications/misc/megasync/default.nix]: Correct, I just the left bracket sitting on its original place one line below the assignment line.

**Comment by michojel** [comment]
@SuperSandro2000 thank you!

**Pull request merged**